analyzer/src/main/kotlin/managers/PackageJsonUtil.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package managers
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.here.ort.model.readValue
+import java.io.File
+import java.nio.file.FileSystems
+import java.nio.file.PathMatcher
+
+internal class PackageJsonUtil {
+    companion object {
+        private val NPM_LOCK_FILES = listOf("npm-shrinkwrap.json", "package-lock.json")
+        private val YARN_LOCK_FILES = listOf("yarn.lock")
+
+        private data class DefinitionFileInfo(
+                val definitionFile: File,
+                var hasYarnLockfile: Boolean = false,
+                var hasNpmLockfile: Boolean = false,
+                var isYarnWorkspaceRoot: Boolean = false,
```

Comment:
I'd like to keep this as is because I think the  value of immutable is relatively low since the data class has very small visibility scope, while I think the current version is a bit more readable which seems more important to me. It would also keep out any logic out of the data class. Can we keep that as is?

Reply:
Ok.